### PR TITLE
[Feature] Add support for token authentication

### PR DESCRIPTION
**Pull request by JunFu0814**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,30 +13,47 @@ If you require features not yet available in this plugin (including client versi
...
+    producer_name => "%{producer_name}"
+    enable_batching => true
+  }
+}
+```
+pulsar with token
+```
+output {
+  pulsar{
+    serviceUrl => "pulsar://10.2.0.7:6650"
```
Comment (hezhangjian) [inline README.md]:
```suggestion
    serviceUrl => "pulsar:/localhost:6650"
```

File: src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    // broker
+    String serviceUrl = "pulsar://127.0.0.1:6650";
+    // topic
+    String topic = "public/default/logstash";
+
+    String token = "eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJhZG1pbiIsImV4cCI6MTY5NDIzNjAwOX0.eFlDpXiiPlC4jeElru0z7NvjwHbmv5eF8orKlr96hSE";
+
+    Map<String, Object> configValues = new HashMap<>();
+
+    // 初始化配置信息
```
Comment (hezhangjian) [inline src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java]: please use english here

File: src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    @Before
+    public void init(){
+
```
Comment (hezhangjian) [inline src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java]: shall we remove the useless blank line?

File: src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import org.logstash.plugins.codecs.Line;
+
+
+public class PulsarOutputExampleTest {
+
+    // broker
+    String serviceUrl = "pulsar://127.0.0.1:6650";
+    // topic
```
Comment (hezhangjian) [inline src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java]: I think we can remove this annotation, it doesn't provide any extra information

File: src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import org.logstash.plugins.codecs.Line;
+
+
+public class PulsarOutputExampleTest {
+
+    // broker
```
Comment (hezhangjian) [inline src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java]: I think we can remove this annotation, it doesn't provide any extra information

File: src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        int eventCount = 5;
+        Collection<Event> events = new ArrayList<>();
+        for (int k = 0; k < eventCount; k++) {
+            Event e = new org.logstash.Event();
+            e.setField(sourceField, "message :  hello jun test " + k);
+            events.add(e);
+        }
+
+        System.out.println("event : " + events);
+        output.output(events);
```
Comment (hezhangjian) [inline src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java]: we can assert the events size. A test case needs assert causes.

File: src/main/java/org/apache/pulsar/logstash/outputs/Pulsar.java [hunk 7]
```diff
@@ -139,10 +146,15 @@ public Pulsar(final String id, final Configuration configuration, final Context
...
         enableTls = configuration.get(CONFIG_ENABLE_TLS);
+        enableToken = configuration.get(CONFIG_ENABLE_TOKEN);
+
         try {
             if (enableTls) {
                 // pulsar TLS
                 client = buildTlsPulsar(configuration);
+            } else if (enableToken) {
+                // pulsar Token
+                client = buildTokenPulsar(configuration);
```
Comment (hezhangjian) [inline src/main/java/org/apache/pulsar/logstash/outputs/Pulsar.java:161]: can token and tls be both enabled?

**Review by hezhangjian — CHANGES_REQUESTED** [review]

**Comment by JunFu0814** [comment]
@Shoothzj please review again

File: src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        for (int k = 0; k < eventCount; k++) {
+            Event e = new org.logstash.Event();
+            e.setField(sourceField, "message : hello test " + k);
+            events.add(e);
+        }
+        output.output(events);
+        Assert.assertEquals("events size is 5", events.size(), eventCount);
+        int index = 0;
+        while (index < eventCount) {
+            Assert.assertTrue("event0 contains the specified str", events.get(index).getField("message").toString().contains("message : hello test " + index));
```
Comment (hezhangjian) [inline src/test/java/org/apache/pulsar/logstash/outputs/PulsarOutputExampleTest.java]: Will this 0 not be incremented with it?

**Comment by hezhangjian** [comment]
Currently, it seems the test-cases not set up the pulsar broker. It is not appropriate to submit test cases at this stage.
@JavaXiaoJun What do you think? How others have run this test?

**Review by hezhangjian — APPROVED** [review]

**Pull request merged**